src/eventbuffer/InMemoryJSONEventBuffer.ts

```diff
@@ -488,46 +432,75 @@ export default class InMemoryJSONEventBuffer implements EventBuffer<EventData>,
           this.logger.debug(
             `Event Reporting - InMemoryJSONEventBuffer - sendEventImmediately - beaconing data out ${body}`
           );
-          /* istanbul ignore else */
           if (!navigator.sendBeacon(`${this.ingestionURL}?beacon=1`, body)) {
             failed = true;
+          } else {
+            failed = false;
           }
         }
       } catch (error) {
-        this.logger.warn(`Event Reporting - Error sending beacon for an important event ${name}`);
+        this.logger.warn(`Event Reporting - Error sending beacon for an important event ${body}`);
         failed = true;
       }
     }
 
     /* istanbul ignore else */
     if (failed) {
       this.logger.debug(
-        `Event Reporting - InMemoryJSONEventBuffer - sendEventImmediately - pushing to failed events ${JSON.stringify(
-          event
-        )}`
+        `Event Reporting - InMemoryJSONEventBuffer - sendEventImmediately - pushing to failed events ${body}`
       );
       this.failedIngestionEvents.push(event);
     }
   }
 
+  private getBackoffWaitTime(retryCount: number): number {
+    return (
+      Math.random() *
+      Math.min(
+        InMemoryJSONEventBuffer.RETRY_LONG_BACKOFF_MS,
+        InMemoryJSONEventBuffer.RETRY_SHORT_BACKOFF_MS * Math.pow(2.0, retryCount)
+      )
+    );
+  }
+
+  private wait(delayInMs: number): Promise<void> {
+    return new Promise(resolve => setTimeout(resolve, delayInMs));
+  }
+
   private async send(data: string): Promise<Response> {
     try {
-      const response = await fetch(this.ingestionURL, {
-        method: 'POST',
-        headers: {
-          Authorization: `Bearer ${this.authenticationToken}`,
-        },
-        body: data,
-      });
-      return response;
+      let retryCount = 0;
+      while (retryCount < this.retryCountLimit) {
```

Comment:
Could we extend the current FullJitterBackoff to handle custom retry count? Seem like useful in other place in the future too.

Reply:
Actually that has to be a static method and as explained in the description I want to use it at two places (sending important events and while sending normal events) hence cannot really create an object to get the different backoff time to use at two places when needed. So I implemented it in this class. Checking if I can extend existing implementation.

Reply:
I checked the implementaion further, and see there are two ways to achieve this. 
1. Add a separate class. This class wont implement the [`Backoff`](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/backoff/Backoff.ts) interface nor extend the existing [`FullJitterBackoff`](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/backoff/FullJitterBackoff.ts) as it actually wont use it. But this would be a re-usable class now.
```ts
export default class FullJitterBackoffWithRetryCount {
  private static FIXED_WAIT_MS = 0;
  private static SHORT_BACKOFF_MS = 0;
  private static LONG_BACKOFF_MS = 0;

  constructor(
    fixedWaitMs: number,
    shortBackoffMs: number,
    longBackoffMs: number
  ) {
    FullJitterBackoffWithRetryCount.FIXED_WAIT_MS = fixedWaitMs;
    FullJitterBackoffWithRetryCount.SHORT_BACKOFF_MS = shortBackoffMs;
    FullJitterBackoffWithRetryCount.LONG_BACKOFF_MS = longBackoffMs;
  }

  static nextBackoffAmountMs(retryCount: number): number {
    const fullJitterMs =
      Math.random() *
        Math.min(this.LONG_BACKOFF_MS, this.SHORT_BACKOFF_MS * Math.pow(2.0, retryCount)) +
      this.FIXED_WAIT_MS;
    return fullJitterMs;
  }
}
```

2. Add an optional `nextRetryIndex` to the Backoff interface and then change the `nextBackoffAmountMs` method implementation to handle this so for any number of objects we would get the needed correct backoff time for a particular retry count/index.
```ts
nextBackoffAmountMs(nextRetryIndex?: number): number;
```

```ts
 nextBackoffAmountMs(nextRetryIndex?: number): number {
    if (nextRetryIndex) {
      this.currentRetry = nextRetryIndex;
    }
    const fullJitterMs =
      Math.random() *
        Math.min(this.longBackoffMs, this.shortBackoffMs * Math.pow(2.0, this.currentRetry)) +
      this.fixedWaitMs;
    this.currentRetry += 1;
    return fullJitterMs;
  }
```

I prefer the first option over existing in-class implementation and the second since we should avoid an optional parameter. Let me know your thoughts on this.

Reply:
Have updated the code to re-use the existing `FullJitterBackoff` class, thanks.